File: ALODWebUtility/Perms/Permission.vb
Namespace Permission

    <Serializable()>
    Public Class Permission

        Protected _allowed As Boolean
        Protected _description As String
        Protected _exclude As Boolean = False
        Protected _id As Integer
        Protected _name As String
        Protected _status As String

        Public Sub New()
            _id = 0
            _name = String.Empty
            _description = String.Empty
        End Sub

        Public Sub New(ByVal id As Integer, ByVal name As String, ByVal description As String)
            _id = id
            _name = name
            _description = description
            _allowed = True
        End Sub

        Public Sub New(ByVal id As Integer, ByVal name As String, ByVal description As String, ByVal allowed As Boolean)
            _id = id
            _name = name
            _description = description
            _allowed = allowed
        End Sub

        Public Property Allowed() As Boolean
            Get
                Return _allowed
            End Get
            Set(ByVal value As Boolean)
                _allowed = value
            End Set
        End Property

        Public Property Description() As String
            Get
                Return _description
            End Get
            Set(ByVal value As String)
                _description = value
            End Set
        End Property

        Public Property Exclude() As Boolean
            Get
                Return _exclude
            End Get
            Set(ByVal value As Boolean)
                _exclude = value
            End Set
        End Property

        Public Property Id() As Integer
            Get
                Return _id
            End Get
            Set(ByVal value As Integer)
                _id = value
            End Set
        End Property

        Public Property Name() As String
            Get
                Return _name
            End Get
            Set(ByVal value As String)
                _name = value
            End Set
        End Property

        Public Property Status() As String
            Get
                Return _status
            End Get
            Set(ByVal value As String)
                _status = value
            End Set
        End Property

        Public Sub ToDataRow(ByVal row As DataSets.PermissionRow)

            row.id = _id
            row.name = _name
            row.description = _description
            row.allowed = _allowed
            row.exclude = _exclude

        End Sub

    End Class

End Namespace

File: ALODWebUtility/Perms/Permission.cs
using System;

namespace ALODWebUtility.Perms
{
    [Serializable]
    public class Permission
    {
        protected bool _allowed;
        protected string _description;
        protected bool _exclude = false;
        protected int _id;
        protected string _name;
        protected string _status;

        public Permission()
        {
            _id = 0;
            _name = string.Empty;
            _description = string.Empty;
        }

        public Permission(int id, string name, string description)
        {
            _id = id;
            _name = name;
            _description = description;
            _allowed = true;
        }

        public Permission(int id, string name, string description, bool allowed)
        {
            _id = id;
            _name = name;
            _description = description;
            _allowed = allowed;
        }

        public bool Allowed
        {
            get
            {
                return _allowed;
            }
            set
            {
                _allowed = value;
            }
        }

        public string Description
        {
            get
            {
                return _description;
            }
            set
            {
                _description = value;
            }
        }

        public bool Exclude
        {
            get
            {
                return _exclude;
            }
            set
            {
                _exclude = value;
            }
        }

        public int Id
        {
            get
            {
                return _id;
            }
            set
            {
                _id = value;
            }
        }

        public string Name
        {
            get
            {
                return _name;
            }
            set
            {
                _name = value;
            }
        }

        public string Status
        {
            get
            {
                return _status;
            }
            set
            {
                _status = value;
            }
        }

        public void ToDataRow(DataSets.PermissionRow row)
        {
            row.id = _id;
            row.name = _name;
            row.description = _description;
            row.allowed = _allowed;
            row.exclude = _exclude;
        }
    }
}
